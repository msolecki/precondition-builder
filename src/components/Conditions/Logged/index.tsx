import React, {ChangeEvent} from 'react'
import {makeStyles} from '@material-ui/core/styles'
import MenuItem from '@material-ui/core/MenuItem'
import FormControl from '@material-ui/core/FormControl'
import Select from '@material-ui/core/Select'
import PropTypes from 'prop-types'
import Title from '../../Title'

const useStyles = makeStyles(({spacing}) => ({
    root: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    formControl: {
        marginTop: spacing(1),
        marginBottom: spacing(3),
        minWidth: 120,
    },
}))

interface LoggedProps {
    onDataChange(value: boolean): void;

    logged: boolean | null;
}

const Logged: React.FC<LoggedProps> = (props: LoggedProps): React.ReactElement => {
    const classes = useStyles()

    function handleChange(event: ChangeEvent<{}>): void {
        const {value} = event.target as HTMLSelectElement

        props.onDataChange(Number(value) === 1)
    }

    return (
        <>
            <Title text="Logged in"/>
            <FormControl className={classes.formControl} fullWidth required component='div'>
                <Select
                    value={props.logged === null ? '' : Number(props.logged)}
                    displayEmpty
                    onChange={handleChange}
                    inputProps={{
                        name: 'activated',
                        id: 'activated',
                    }}
                >
                    <MenuItem component='li' button={true} value={1}>True</MenuItem>
                    <MenuItem component='li' button={true} value={0}>False</MenuItem>
                </Select>
            </FormControl>
        </>
    )
}

Logged.propTypes = {
    onDataChange: PropTypes.func.isRequired,
    logged: PropTypes.bool,
}

Logged.defaultProps = {
    logged: null,
}

export default Logged
