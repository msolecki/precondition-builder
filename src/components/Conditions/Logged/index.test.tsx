import * as React from 'react'
import {ReactTestRenderer} from 'react-test-renderer'
import {createShallow} from '@material-ui/core/test-utils'

import Logged, {LoggedProps} from './index'
import {Select} from '@material-ui/core'

const component = (props: LoggedProps): Logged => <Logged {...props}/>
const onDataChangeFunc = jest.fn()

describe('<Logged />', () => {
    let shallow

    beforeAll(() => {
        shallow = createShallow()
    })

    test('empty snapshot', () => {
        const wrapper: ReactTestRenderer = shallow(
            component({
                logged: null,
                onDataChange: () => {
                }
            }))

        expect(wrapper).toMatchSnapshot()
    })

    test('default snapshot', () => {
        const wrapper = shallow(
            component({
                logged: true,
                onDataChange: onDataChangeFunc
            }))
        const event = {target: {name: 'name', value: 1}}

        wrapper.find(Select).simulate('change', event)

        expect(onDataChangeFunc).toHaveBeenCalledWith(true)
    })
})
