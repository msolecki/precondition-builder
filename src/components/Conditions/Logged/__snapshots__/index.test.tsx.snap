// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logged /> empty snapshot 1`] = `
<Fragment>
  <Title
    text="Logged in"
  />
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-formControl-2"
    component="div"
    fullWidth={true}
    required={true}
  >
    <WithStyles(WithFormControlContext(ForwardRef(Select)))
      displayEmpty={true}
      inputProps={
        Object {
          "id": "activated",
          "name": "activated",
        }
      }
      onChange={[Function]}
      value=""
    >
      <WithStyles(ForwardRef(MenuItem))
        button={true}
        component="li"
        value={1}
      >
        True
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        button={true}
        component="li"
        value={0}
      >
        False
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(WithFormControlContext(ForwardRef(Select)))>
  </WithStyles(ForwardRef(FormControl))>
</Fragment>
`;
